### DISCUSSION TITLE
generate waveform from audio files?

### DISCUSSION THREAD

**Discussion by jayenne** [spatie/laravel-medialibrary#2477] [Ideas]
Hi, is it possible to generate a waveform image/thumb for audio files as can be done with videos?

IF notm, then in the interest of completness, it would be nice to be able to generate thumbs for each media type and a such include audio too.

the waveform could has a 'resolution' argument that samples at different resolutions for small simple waveform image or larger more details looking images.
reference: [https://justwave.beotiger.com/index.html](https://justwave.beotiger.com/index.html)
It might be nice generate a short 'clip' video, either a clip between two timestamps or every n frames between two timestamps.

**Comment by freekmurze** [comment]
Out of the box, the media library doesn't support generating waveform images.

@Synchro and @Gummibeer have been working with audio recently. Maybe they can help you out.

**Reply by Synchro** [reply]
Thanks @freekmurze! The thing you're looking for is my [laravel-medialibrary-audio](https://github.com/Synchro/laravel-medialibrary-audio). I also have support working for generating spectra instead of waveforms (they're much prettier!), but my PR in PHP-FFMPEG to allow it is currently languishing unmerged.
